# 1. Importación de datos


In [1]:
import pandas as pd

#Leemos los archivos del repositorio
url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

#Cargamos y etiquetamos los datos obtenidos de los archivos csv. Identificamos el origen: agregamos una columna "Tienda" que indique de qué archivo proviene cada registro.
tienda1 = pd.read_csv(url);  tienda1["Tienda"] = "Tienda 1"
tienda2 = pd.read_csv(url2); tienda2["Tienda"] = "Tienda 2"
tienda3 = pd.read_csv(url3); tienda3["Tienda"] = "Tienda 3"
tienda4 = pd.read_csv(url4); tienda4["Tienda"] = "Tienda 4"

#Unimos los datos de las tiendas
tiendas = pd.concat([tienda1, tienda2, tienda3, tienda4], ignore_index=True)

#convertimos las fechas para su uso como datos
tiendas["Fecha de Compra"] = pd.to_datetime(tiendas["Fecha de Compra"], dayfirst=True)

#Verificamos la estructura de neustro nuevo dataframe
tiendas.info()
tiendas.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9435 entries, 0 to 9434
Data columns (total 13 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   Producto                9435 non-null   object        
 1   Categoría del Producto  9435 non-null   object        
 2   Precio                  9435 non-null   float64       
 3   Costo de envío          9435 non-null   float64       
 4   Fecha de Compra         9435 non-null   datetime64[ns]
 5   Vendedor                9435 non-null   object        
 6   Lugar de Compra         9435 non-null   object        
 7   Calificación            9435 non-null   int64         
 8   Método de pago          9435 non-null   object        
 9   Cantidad de cuotas      9435 non-null   int64         
 10  lat                     9435 non-null   float64       
 11  lon                     9435 non-null   float64       
 12  Tienda                  9435 non-null   object  

Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      2021-01-16      Pedro Gomez          Bogotá             4   
1      2022-05-18  Beatriz Morales        Medellín             1   
2      2021-03-15   Juan Fernandez       Cartagena             1   
3      2022-05-03   Juan Fernandez            Cali             4   
4      2020-11-07    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon    Tienda  
0  Tarjeta de crédito                   8   4.60971 -74.08175  Tienda 1  
1  Tarjeta de crédito                   4   6.25184 -75.56359  Tienda 1  
2  Tarjeta de crédito                   1  10.39972 -75.51444  Tienda 1  
3               Nequi                   1   3.43722 -76.52250  Tienda 1  
4               Nequi                   1   6.25184 -75.56359  Tienda 1

# 2. Análisis exploratorio de los datos(EDA)

### Comprendiendo la estructura de los datos

Descripción: Antes de pasar a análisis detallados, es esencial explorar el conjunto de datos para comprender su estructura y contenido. Este paso le permite identificar patrones, inconsistencias y las columnas más relevantes para los siguientes pasos.

**Estructura de datos:**

El conjunto de datos incluye la siguiente información:

    Producto y Categoría: Artículos vendidos y sus calificaciones.

    Precio y Envío: Valores de venta y costos asociados.

    Fecha y ubicación de compra: Información temporal y geográfica.

    Evaluación de compra: Comentarios de clientes.

    Tipo de Pago y Cuotas: Métodos utilizados por los clientes.

    Coordenadas Geográficas: Ubicación de las transacciones.

#### **Explorando carfacterísticas básicas**

In [2]:
#Información general
tiendas.info()

#Estadísticas básicas (sólo núericas)
tiendas.describe()

#Valores únicos por columna(será útil por categoria)
tiendas.nunique()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9435 entries, 0 to 9434
Data columns (total 13 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   Producto                9435 non-null   object        
 1   Categoría del Producto  9435 non-null   object        
 2   Precio                  9435 non-null   float64       
 3   Costo de envío          9435 non-null   float64       
 4   Fecha de Compra         9435 non-null   datetime64[ns]
 5   Vendedor                9435 non-null   object        
 6   Lugar de Compra         9435 non-null   object        
 7   Calificación            9435 non-null   int64         
 8   Método de pago          9435 non-null   object        
 9   Cantidad de cuotas      9435 non-null   int64         
 10  lat                     9435 non-null   float64       
 11  lon                     9435 non-null   float64       
 12  Tienda                  9435 non-null   object  

Producto                    51
Categoría del Producto       8
Precio                    5483
Costo de envío            1204
Fecha de Compra           1185
Vendedor                    14
Lugar de Compra             19
Calificación                 5
Método de pago               4
Cantidad de cuotas          19
lat                         19
lon                         19
Tienda                       4
dtype: int64

#### **Comprobación de valores nulos, faltantes o inconsistencias**

In [3]:
#Contamos los valores nulos por columna
tiendas.isnull().sum()

#Verificamos si hay duplicados
tiendas.duplicated().sum()

np.int64(0)

#3. Análisis de facturación



#### **Promedio o número de ventas de artículos de cada tienda y volumen total de transacciones**

In [4]:
#promedio o número de ventas de artículos de cada tienda y volumen total de transacciones
tiendas.groupby("Tienda")["Producto"].count()


Tienda
Tienda 1    2359
Tienda 2    2359
Tienda 3    2359
Tienda 4    2358
Name: Producto, dtype: int64

#### **Promedio y total de ingresos( Precio+envío) de cada tienda**

In [5]:
#Identificamos el promedio y total de ingresos( Precio+envío) de cada tienda
#mostramos la media del número de ventas totales de cada tienda y la ganancia total de todas al ventas de cada tienda.
ventas = tiendas.groupby("Tienda")["Precio"].sum().reset_index()
ventas["Precio_COP"] = ventas["Precio"].apply(lambda x:f"${x:,.0f} COP")

#Mostramos también en dolares
TASA_USD = 3865
ventas["Precio_USD"] = ventas["Precio"] /TASA_USD
ventas["Precio_USD"] = ventas["Precio_USD"].apply(lambda x:f"${x:,.2f} USD")
ventas.drop("Precio", axis=1, inplace=True)
ventas


Tienda          Precio_COP       Precio_USD
0  Tienda 1  $1,150,880,400 COP  $297,769.83 USD
1  Tienda 2  $1,116,343,500 COP  $288,834.02 USD
2  Tienda 3  $1,098,019,600 COP  $284,093.04 USD
3  Tienda 4  $1,038,375,700 COP  $268,661.24 USD

#### **Diversidad o preferencia de los métodos de pago de los compradores**

In [6]:
#identificamos la diversidad o preferencia de los métodos de pago de los compradores
tiendas.groupby("Tienda")["Método de pago"].value_counts()

Tienda    Método de pago    
Tienda 1  Tarjeta de crédito    1704
          Nequi                  483
          Cupón                  137
          Tarjeta de débito       35
Tienda 2  Tarjeta de crédito    1721
          Nequi                  467
          Cupón                  141
          Tarjeta de débito       30
Tienda 3  Tarjeta de crédito    1753
          Nequi                  468
          Cupón                  101
          Tarjeta de débito       37
Tienda 4  Tarjeta de crédito    1720
          Nequi                  482
          Cupón                  125
          Tarjeta de débito       31
Name: count, dtype: int64

# 4. Ventas por categoría

#### **Número de artículos vendidos por categoria de productos de cada una de las tiendas**

In [7]:
#Vamos a mostrar las ventas por categoria de productos de cada una de las tiendas
tiendas.groupby(["Tienda", "Categoría del Producto"])["Producto"].count()


Tienda    Categoría del Producto 
Tienda 1  Artículos para el hogar    171
          Deportes y diversión       284
          Electrodomésticos          312
          Electrónicos               448
          Instrumentos musicales     182
          Juguetes                   324
          Libros                     173
          Muebles                    465
Tienda 2  Artículos para el hogar    181
          Deportes y diversión       275
          Electrodomésticos          305
          Electrónicos               422
          Instrumentos musicales     224
          Juguetes                   313
          Libros                     197
          Muebles                    442
Tienda 3  Artículos para el hogar    177
          Deportes y diversión       277
          Electrodomésticos          278
          Electrónicos               451
          Instrumentos musicales     177
          Juguetes                   315
          Libros                     185
          Muebles                    499
Tienda 4  Artículos para el hogar    201
          Deportes y diversión       277
          Electrodomésticos          254
          Electrónicos               451
          Instrumentos musicales     170
          Juguetes                   338
          Libros                     187
          Muebles                    480
Name: Producto, dtype: int64

#### **Valor total de la venta de productos por categoria por cada una de las 4 tiendas**

In [8]:
#Mostramos el valor de la venta por categoria de los productos por cada una de las 4 tiendas
GanPorTienda = tiendas.groupby(["Tienda", "Categoría del Producto"])["Precio"].sum()
GanPorTienda

Tienda    Categoría del Producto 
Tienda 1  Artículos para el hogar     12698400.0
          Deportes y diversión        39290000.0
          Electrodomésticos          363685200.0
          Electrónicos               429493500.0
          Instrumentos musicales      91299000.0
          Juguetes                    17995700.0
          Libros                       8784900.0
          Muebles                    187633700.0
Tienda 2  Artículos para el hogar     14746900.0
          Deportes y diversión        34744500.0
          Electrodomésticos          348567800.0
          Electrónicos               410831100.0
          Instrumentos musicales     104990300.0
          Juguetes                    15945400.0
          Libros                      10091200.0
          Muebles                    176426300.0
Tienda 3  Artículos para el hogar     15060000.0
          Deportes y diversión        35593100.0
          Electrodomésticos          329237900.0
          Electrónicos               410775800.0
          Instrumentos musicales      77380900.0
          Juguetes                    19401100.0
          Libros                       9498700.0
          Muebles                    201072100.0
Tienda 4  Artículos para el hogar     15074500.0
          Deportes y diversión        33350100.0
          Electrodomésticos          283260200.0
          Electrónicos               409476100.0
          Instrumentos musicales      75102400.0
          Juguetes                    20262200.0
          Libros                       9321300.0
          Muebles                    192528900.0
Name: Precio, dtype: float64

#### **Valor total de la venta de productos por categoria en COP/USD por cada una de las 4 tiendas**

In [9]:
import plotly.express as px

# Mostramos las ganancias en pesos y dólares de cada categoria por tienda
GanPorTienda = tiendas.groupby(["Tienda", "Categoría del Producto"])["Precio"].sum().reset_index()

# Creamos un gráfico de barras
fig = px.bar(GanPorTienda,
             x="Categoría del Producto",
             y="Precio",
             color="Tienda",
             barmode="group", # Agrupar barras por categoría
             title="Valor Total de Venta por Categoría y Tienda (COP)")

# Ajustamos el diseño del gráfico
fig.update_layout(
    xaxis={'categoryorder':'total descending'}, # Ordenar categorías por el valor total
    yaxis=dict(
        tickformat=",.0f", # Formato para pesos colombianos sin decimales
        hoverformat=",.0f" # Formato para hover
    ),
    hovermode="x unified"
)

fig.show()

# Mostramos la tabla formateada en COP y USD también
GanPorTienda["Precio_COP"] = GanPorTienda["Precio"].apply(lambda x:f"${x:,.0f} COP")
TASA_USD = 3865
GanPorTienda["Precio_USD"] = GanPorTienda["Precio"] /TASA_USD
GanPorTienda["Precio_USD"] = GanPorTienda["Precio_USD"].apply(lambda x:f"${x:,.2f} USD")
GanPorTienda.drop("Precio", axis=1, inplace=True)

display(GanPorTienda)

Tienda   Categoría del Producto        Precio_COP       Precio_USD
0   Tienda 1  Artículos para el hogar   $12,698,400 COP    $3,285.49 USD
1   Tienda 1     Deportes y diversión   $39,290,000 COP   $10,165.59 USD
2   Tienda 1        Electrodomésticos  $363,685,200 COP   $94,097.08 USD
3   Tienda 1             Electrónicos  $429,493,500 COP  $111,123.80 USD
4   Tienda 1   Instrumentos musicales   $91,299,000 COP   $23,621.99 USD
5   Tienda 1                 Juguetes   $17,995,700 COP    $4,656.07 USD
6   Tienda 1                   Libros    $8,784,900 COP    $2,272.94 USD
7   Tienda 1                  Muebles  $187,633,700 COP   $48,546.88 USD
8   Tienda 2  Artículos para el hogar   $14,746,900 COP    $3,815.50 USD
9   Tienda 2     Deportes y diversión   $34,744,500 COP    $8,989.52 USD
10  Tienda 2        Electrodomésticos  $348,567,800 COP   $90,185.72 USD
11  Tienda 2             Electrónicos  $410,831,100 COP  $106,295.24 USD
12  Tienda 2   Instrumentos musicales  $104,990,300 COP   $27,164.37 USD
13  Tienda 2                 Juguetes   $15,945,400 COP    $4,125.59 USD
14  Tienda 2                   Libros   $10,091,200 COP    $2,610.92 USD
15  Tienda 2                  Muebles  $176,426,300 COP   $45,647.17 USD
16  Tienda 3  Artículos para el hogar   $15,060,000 COP    $3,896.51 USD
17  Tienda 3     Deportes y diversión   $35,593,100 COP    $9,209.08 USD
18  Tienda 3        Electrodomésticos  $329,237,900 COP   $85,184.45 USD
19  Tienda 3             Electrónicos  $410,775,800 COP  $106,280.93 USD
20  Tienda 3   Instrumentos musicales   $77,380,900 COP   $20,020.93 USD
21  Tienda 3                 Juguetes   $19,401,100 COP    $5,019.69 USD
22  Tienda 3                   Libros    $9,498,700 COP    $2,457.62 USD
23  Tienda 3                  Muebles  $201,072,100 COP   $52,023.83 USD
24  Tienda 4  Artículos para el hogar   $15,074,500 COP    $3,900.26 USD
25  Tienda 4     Deportes y diversión   $33,350,100 COP    $8,628.75 USD
26  Tienda 4        Electrodomésticos  $283,260,200 COP   $73,288.54 USD
27  Tienda 4             Electrónicos  $409,476,100 COP  $105,944.66 USD
28  Tienda 4   Instrumentos musicales   $75,102,400 COP   $19,431.41 USD
29  Tienda 4                 Juguetes   $20,262,200 COP    $5,242.48 USD
30  Tienda 4                   Libros    $9,321,300 COP    $2,411.72 USD
31  Tienda 4                  Muebles  $192,528,900 COP   $49,813.43 USD

# 6. Productos más y menos vendidos

#### **Categoría de productos más vendidos y su valor en COP/USD por cada tienda**

In [10]:
#¿cuál es al categoria de productos que genera más ganancias a EN PESOS Y DÓLARES y el segundo producto que más genera para cada tienda?
GanPorTienda_numeric = tiendas.groupby(["Tienda", "Categoría del Producto"])["Precio"].sum().reset_index()
TASA_USD = 3865 # valor de tasa de cambio de dolar fijada

# Aplicamos nlargest al valor numérico de la columna 'Precio'
top_earning_categories = GanPorTienda_numeric.groupby("Tienda")["Precio"].nlargest(2).reset_index()

# Combinamos GanPorTienda_numeric  para obetener el nombre de la categoria
top_earning_categories_formatted = top_earning_categories.merge(GanPorTienda_numeric[['Tienda', 'Categoría del Producto', 'Precio']],on=['Tienda', 'Precio'])

# Formatemos para observan mejor en pantalla en pesos colombianos y dólares
top_earning_categories_formatted['Precio_COP'] = top_earning_categories_formatted["Precio"].apply(lambda x: f"${x:,.0f} COP")
top_earning_categories_formatted['Precio_USD'] = top_earning_categories_formatted["Precio"].apply(lambda x: f"${x/TASA_USD:,.2f} USD")

# Quitamos la columna precio 'Precio' para visualización
top_earning_categories_formatted.drop('Precio', axis=1, inplace=True)
top_earning_categories_formatted.drop('level_1', axis=1, inplace=True)

top_earning_categories_formatted

Tienda Categoría del Producto        Precio_COP       Precio_USD
0  Tienda 1           Electrónicos  $429,493,500 COP  $111,123.80 USD
1  Tienda 1      Electrodomésticos  $363,685,200 COP   $94,097.08 USD
2  Tienda 2           Electrónicos  $410,831,100 COP  $106,295.24 USD
3  Tienda 2      Electrodomésticos  $348,567,800 COP   $90,185.72 USD
4  Tienda 3           Electrónicos  $410,775,800 COP  $106,280.93 USD
5  Tienda 3      Electrodomésticos  $329,237,900 COP   $85,184.45 USD
6  Tienda 4           Electrónicos  $409,476,100 COP  $105,944.66 USD
7  Tienda 4      Electrodomésticos  $283,260,200 COP   $73,288.54 USD

#### **Categoría de productos menos vendidos y su valor en COP/USD por cada tienda**

In [11]:
#¿cuál es al categoria de productos que genera menos ganancias a EN PESOS Y DÓLARES y el segundo producto que más genera para cada tienda?
GanPorTienda_numeric = tiendas.groupby(["Tienda", "Categoría del Producto"])["Precio"].sum().reset_index()
TASA_USD = 3865 # valor de tasa de cambio de dolar fijada

# Aplicamos nlargest al valor numérico de la columna 'Precio'
low_earning_categories = GanPorTienda_numeric.groupby("Tienda")["Precio"].nsmallest(2).reset_index()

# Combinamos GanPorTienda_numeric  para obetener el nombre de la categoria
low_earning_categories_formatted = low_earning_categories.merge(GanPorTienda_numeric[['Tienda', 'Categoría del Producto', 'Precio']],on=['Tienda', 'Precio'])

# Formatemos para observan mejor en pantalla en pesos colombianos y dólares
low_earning_categories_formatted['Precio_COP'] = low_earning_categories_formatted["Precio"].apply(lambda x: f"${x:,.0f} COP")
low_earning_categories_formatted['Precio_USD'] = low_earning_categories_formatted["Precio"].apply(lambda x: f"${x/TASA_USD:,.2f} USD")

# Quitamos la columna precio 'Precio' para visualización
low_earning_categories_formatted.drop('Precio', axis=1, inplace=True)
low_earning_categories_formatted.drop('level_1', axis=1, inplace=True)

low_earning_categories_formatted

Tienda   Categoría del Producto       Precio_COP     Precio_USD
0  Tienda 1                   Libros   $8,784,900 COP  $2,272.94 USD
1  Tienda 1  Artículos para el hogar  $12,698,400 COP  $3,285.49 USD
2  Tienda 2                   Libros  $10,091,200 COP  $2,610.92 USD
3  Tienda 2  Artículos para el hogar  $14,746,900 COP  $3,815.50 USD
4  Tienda 3                   Libros   $9,498,700 COP  $2,457.62 USD
5  Tienda 3  Artículos para el hogar  $15,060,000 COP  $3,896.51 USD
6  Tienda 4                   Libros   $9,321,300 COP  $2,411.72 USD
7  Tienda 4  Artículos para el hogar  $15,074,500 COP  $3,900.26 USD

#### **Productos más y menos vendidos vendidos y su valor en COP/USD por cada tienda**

In [12]:
#Queremos saber los pruductos más vendidos y menos vendidos por cada tienda con su valor en COP y USD
ganancia_por_producto = tiendas.groupby(["Tienda", "Producto"])["Precio"].sum()
TASA_USD = 3865 # Assuming this is already defined

# Obtenemos un productos de cada tienda que sea el más vendido
top_products = ganancia_por_producto.groupby("Tienda").nlargest(1).reset_index(level=1)
top_products.columns = ['Producto', 'Precio']
top_products["Precio_COP"] = top_products["Precio"].apply(lambda x: f"${x:,.0f} COP")
top_products["Precio_USD"] = top_products["Precio"].apply(lambda x: f"${x/TASA_USD:,.2f} USD")
top_products.drop("Precio", axis=1, inplace=True)
print("Top productos mas vendidos por tienda:")
display(top_products)

# Obtenemos un productos de cada tienda que sea el menos vendido
bottom_products = ganancia_por_producto.groupby("Tienda").nsmallest(1).reset_index(level=1)
bottom_products.columns = ['Producto', 'Precio']
bottom_products["Precio_COP"] = bottom_products["Precio"].apply(lambda x: f"${x:,.0f} COP")
bottom_products["Precio_USD"] = bottom_products["Precio"].apply(lambda x: f"${x/TASA_USD:,.2f} USD")
bottom_products.drop("Precio", axis=1, inplace=True)
print("\nTop productos menos vendidos por tienda:")
display(bottom_products)

Top productos mas vendidos por tienda:


Producto        Precio_COP      Precio_USD
Tienda   Producto                                                 
Tienda 1 TV LED UHD 4K  Tienda 1  $142,150,200 COP  $36,778.84 USD
Tienda 2 Iphone 15      Tienda 2  $113,225,800 COP  $29,295.16 USD
Tienda 3 Refrigerador   Tienda 3   $99,889,700 COP  $25,844.68 USD
Tienda 4 Iphone 15      Tienda 4   $96,697,500 COP  $25,018.76 USD


Top productos menos vendidos por tienda:


Producto    Precio_COP   Precio_USD
Tienda   Producto                                               
Tienda 1 Cuerda para saltar  Tienda 1  $652,800 COP  $168.90 USD
Tienda 2 Cubo mágico 8x8     Tienda 2  $643,600 COP  $166.52 USD
Tienda 3 Cubo mágico 8x8     Tienda 3  $640,400 COP  $165.69 USD
Tienda 4 Cuerda para saltar  Tienda 4  $704,800 COP  $182.35 USD

# 5. Calificación promedio de la tienda


#### **Calificación promedio de cada tienda**

In [13]:
#Vamos a mostra rl promedio de las calificaciones de los clientes por tienda formateado a dos decimales
tiendas.groupby("Tienda")["Calificación"].mean().apply(lambda x:f"{x:,.2f}")

Tienda
Tienda 1    3.98
Tienda 2    4.04
Tienda 3    4.05
Tienda 4    4.00
Name: Calificación, dtype: object

#### **Volumen de compra de las ciudades**

In [14]:
# queremos visualizar las ciudades que más compran o de que ciudades vienen más volumen de compra
ciudades_mas_compra = tiendas["Lugar de Compra"].value_counts().reset_index(name="Cantidad")
ciudades_mas_compra.rename(columns={"index": "Lugar de Compra"}, inplace=True)
display(ciudades_mas_compra)

Lugar de Compra  Cantidad
0           Bogotá      3943
1         Medellín      2314
2             Cali      1007
3          Pereira       517
4     Barranquilla       310
5        Cartagena       242
6           Cúcuta       186
7      Bucaramanga       181
8         Riohacha       163
9      Santa Marta       151
10         Leticia       142
11           Pasto        94
12       Manizales        41
13           Neiva        40
14   Villavicencio        29
15         Armenia        28
16          Soacha        19
17         Inírida        14
18      Valledupar        14

#### **Volumen de compra por ciudad para cada tienda**

In [15]:
# queremos saber cual es el volumen de compra que reciben de cada ciudad cada una de las tiendas y expresarlo en un gráfico
import plotly.express as px

# Agrupamos los datos por ciudad y tienda para contar el número de productos vendidos
volumen_compra_por_ciudad_tienda = tiendas.groupby(["Lugar de Compra", "Tienda"])["Producto"].count().reset_index(name="Cantidad de Artículos Vendidos")

# Creamos el gráfico de barras utilizando Plotly Express
fig = px.bar(volumen_compra_por_ciudad_tienda,
             x="Lugar de Compra",
             y="Cantidad de Artículos Vendidos",
             color="Tienda", # Diferenciamos las barras por tienda
             barmode="group", # Agrupamos las barras por ciudad
             title="Volumen de Compra por Ciudad y Tienda")

# Mejoramos la visualización del gráfico
fig.update_layout(xaxis={'categoryorder':'total descending'}) # Ordenamos las ciudades por la cantidad total vendida
fig.show()

# 7. Envío promedio por tienda

#### **Costo promedio de envío por tienda**

In [16]:
#Queremos saber el costo de envío por tienda
envio_por_tienda = tiendas.groupby("Tienda")["Costo de envío"].mean().reset_index()
TASA_USD = 3865  # Assuming this is already defined

envio_por_tienda["Costo de envío COP"] = envio_por_tienda["Costo de envío"].apply(lambda x: f"${x:,.2f} COP")
envio_por_tienda["Costo de envío USD"] = envio_por_tienda["Costo de envío"].apply(lambda x: f"${x/TASA_USD:,.2f} USD")

envio_por_tienda.drop("Costo de envío", axis=1, inplace=True)

display(envio_por_tienda)

Tienda Costo de envío COP Costo de envío USD
0  Tienda 1     $26,018.61 COP          $6.73 USD
1  Tienda 2     $25,216.24 COP          $6.52 USD
2  Tienda 3     $24,805.68 COP          $6.42 USD
3  Tienda 4     $23,459.46 COP          $6.07 USD

#### **Distribución de productos más vendidos por cada tienda en Colombia**

In [17]:
#Queremos saber utilizando la latitud y longitud que tenemos como se distribuye en un mapa de Colombia los productos de las categorías más vendidas por tienda.
import plotly.express as px


#Obetenemos el los datos de las categorias que mas venden del anáisis anteriores
top_categories_list = top_earning_categories_formatted['Categoría del Producto'].unique()

#filtramos el dataset e incluimos solo las categorias del producto que son top en ganacias
tiendas_top_categories = tiendas[tiendas['Categoría del Producto'].isin(top_categories_list)]

#Creamos una visualización del mapa para cada tienda
fig = px.scatter_geo(tiendas_top_categories,
                     lat='lat',
                     lon='lon',
                     color='Categoría del Producto', # Color by category within each store map
                     hover_name='Producto',
                     size='Precio',
                     projection='natural earth',
                     scope='south america',
                     facet_col='Tienda', # Create a separate map for each store
                     title='Distribución Geográfica de las Categorías de Productos Más Vendidas por Tienda')

fig.update_geos(
    visible=False, resolution=110, scope="south america",
    showcountries=True, countrycolor="Black",
    showsubunits=True, subunitcolor="Blue"
)

# mejoramos el foco y el tamaño de cada mapara para Colombia
fig.update_layout(
    geo = dict(
        center=dict(lat=4.5709, lon=-74.2973), # Approximate center of Colombia
        projection_scale=5 # Adjust scale to zoom into Colombia
    ),
    height=700, # Incrementamos altura
    width=1200   # Incrementamos ancho
)
# Para aplicar a todos los mapas las configuraciones
for i in range(1, 5):
    fig.update_layout({
        f'geo{i}': dict(
            center=dict(lat=4.5709, lon=-74.2973),
            projection_scale=5,
            scope='south america'
        )
    })


fig.show()

# 8. Análsis de vendedores

#### **Vendedor con mayor volumen de articulos vendidos, mejor calificado, peor calificado por tienda**

In [18]:
# Queremos ver el vendedor con mayor volumen de articulos vendidos por tienda,
# el de mejor calificación, y el de peor calificación.

#Calculamos el el total de artículos vendidos y el promedio de calificacion para cada vendedor por tienda
vendor_sales_summary = tiendas.groupby(["Tienda", "Vendedor"]).agg(
    Total_Articulos_Vendidos=('Producto', 'count'),
    Promedio_Calificación=('Calificación', 'mean')
).reset_index()

#Encotramos el vendedor con el mayor volumen de articulos vendidos en cada tienda
top_volume_vendor = vendor_sales_summary.loc[vendor_sales_summary.groupby('Tienda')['Total_Articulos_Vendidos'].idxmax()]
print("El vendedor con la mayor cantidad de articulos vendidos:")
display(top_volume_vendor)

# Encotnramos el vendedor con la mejor calificación
best_rated_vendor = vendor_sales_summary.loc[vendor_sales_summary.groupby('Tienda')['Promedio_Calificación'].idxmax()]
print("\nVendedor con la mejor calificación por tienda:")
display(best_rated_vendor)

# Encotnramos el vendedor peor calificado por tienda
worst_rated_vendor = vendor_sales_summary.loc[vendor_sales_summary.groupby('Tienda')['Promedio_Calificación'].idxmin()]
print("\nVendedor peor calificado por tienda:")
display(worst_rated_vendor)

El vendedor con la mayor cantidad de articulos vendidos:


Tienda         Vendedor  Total_Articulos_Vendidos  Promedio_Calificación
4   Tienda 1  Izabela de León                       197               3.751269
18  Tienda 2  Izabela de León                       195               4.000000
36  Tienda 3    Maria Alfonso                       186               3.951613
45  Tienda 4    Felipe Santos                       196               4.015306


Vendedor con la mejor calificación por tienda:


Tienda         Vendedor  Total_Articulos_Vendidos  Promedio_Calificación
13  Tienda 1  Ángel Rodriguez                       169               4.248521
23  Tienda 2  Mariana Herrera                       157               4.191083
28  Tienda 3  Beatriz Morales                       158               4.208861
54  Tienda 4   Santiago Silva                       174               4.160920


Vendedor peor calificado por tienda:


Tienda         Vendedor  Total_Articulos_Vendidos  Promedio_Calificación
4   Tienda 1  Izabela de León                       197               3.751269
15  Tienda 2   Blanca Ramirez                       166               3.933735
32  Tienda 3  Izabela de León                       171               3.730994
52  Tienda 4      Pedro Gomez                       154               3.857143

#### **Reivisón caso Izabela de León**

In [19]:
# Filtramos las ventas hechas porIzabela de León
izabela_sales = tiendas[tiendas['Vendedor'] == 'Izabela de León']

# ordenamos en orden descendente para ver las ventas mas recientes
izabela_recent_sales = izabela_sales.sort_values(by='Fecha de Compra', ascending=False)

# Mostramos las primeras 10 ventas hechas más recientes
print("Ventas más recientes hechas Izabela de León:")
display(izabela_recent_sales[['Fecha de Compra', 'Tienda', 'Producto', 'Precio', 'Calificación']].head(10))

# Ordenamos por calificación en orden ascendente para ver las ventas peor calificadas
izabela_worst_rated_sales = izabela_sales.sort_values(by='Calificación', ascending=True)

# Mostramos las 10 ventas peor calificadas
print("\n10 peores calificadas ventas hechas por Izabela de León:")
display(izabela_worst_rated_sales[['Fecha de Compra', 'Tienda', 'Producto', 'Precio', 'Calificación']].head(10))

Ventas más recientes hechas Izabela de León:


Fecha de Compra    Tienda                     Producto     Precio  \
67        2023-03-31  Tienda 1          Modelado predictivo    53000.0   
912       2023-03-30  Tienda 1                 Vaso térmico    27000.0   
6485      2023-03-27  Tienda 3      Dashboards con Power BI    36400.0   
2136      2023-03-26  Tienda 1                  Auriculares   135900.0   
8068      2023-03-25  Tienda 4                       Sillón   247800.0   
4606      2023-03-25  Tienda 2                       Estufa   389700.0   
6875      2023-03-21  Tienda 3                      Batería  1095600.0   
4706      2023-03-21  Tienda 2                    Cama king   488000.0   
199       2023-03-20  Tienda 1            Ajedrez de madera    37500.0   
707       2023-03-15  Tienda 1  Ciencia de datos con Python    58700.0   

      Calificación  
67               3  
912              3  
6485             1  
2136             5  
8068             5  
4606             5  
6875             3  
4706             4  
199              5  
707              5


10 peores calificadas ventas hechas por Izabela de León:


Fecha de Compra    Tienda                   Producto     Precio  \
8         2022-01-08  Tienda 1           Lavadora de ropa  1518200.0   
9333      2021-04-23  Tienda 4                  Pandereta    88600.0   
9410      2022-02-09  Tienda 4               Lavavajillas  1340600.0   
9315      2022-03-09  Tienda 4                Auriculares   116600.0   
9273      2021-07-28  Tienda 4                Tablet ABXY   668900.0   
272       2021-02-06  Tienda 1               Refrigerador  2406400.0   
610       2022-10-25  Tienda 1               Set de ollas    79700.0   
8589      2022-06-18  Tienda 4  Carrito de control remoto    75800.0   
4484      2022-02-14  Tienda 2               Vaso térmico    54100.0   
4485      2020-01-11  Tienda 2                    Armario   517300.0   

      Calificación  
8                1  
9333             1  
9410             1  
9315             1  
9273             1  
272              1  
610              1  
8589             1  
4484             1  
4485             1

#### **Ventas y calificación para cada vendedor a través de todas las tiendas y su rendimiento**

In [ ]:
with open('README.md', 'r') as f:
    readme_content = f.read()
print(readme_content)

## Resumen:

### Preguntas y Respuestas

1.  **¿Qué tienda, si alguna, debería ser considerada para el cierre basándose en varias métricas financieras y operativas?**
    Basado en el análisis, **Tienda 4** debería ser considerada para el cierre. Aunque la satisfacción del cliente y el volumen de ventas son similares en todas las tiendas, la Tienda 4 muestra una tendencia descendente constante en las ventas y una menor rentabilidad en los últimos años (2022-2023), no logrando alcanzar consistentemente la marca de 30 millones de COP/mes en ingresos que otras tiendas mantienen o superan.

2.  **¿Cuáles son las conclusiones finales y recomendaciones para el Sr. Joao, incluyendo sugerencias para mejorar la recopilación de datos?**
    La conclusión principal es considerar el cierre de la Tienda 4 debido a su tendencia de ingresos decreciente. Sin embargo, una reorganización estratégica de productos también podría ser una opción para mantener las cuatro tiendas activas.

    Recomendaciones para mejorar el registro de datos:
    *   **Diferenciar los Canales de Venta:** Registrar claramente si las compras se realizan en persona o en línea para identificar las preferencias del cliente y optimizar la logística.
    *   **Rastrear los Costos Operativos:** Implementar un sistema para registrar los gastos de mantenimiento, nómina y otros gastos operativos por tienda para permitir el cálculo del beneficio neto real y evaluar el rendimiento verdadero.

### Hallazgos Clave del Análisis de Datos

*   **Calidad de los Datos:** El conjunto de datos es notablemente limpio, sin valores nulos ni registros duplicados en los cuatro conjuntos de datos de tiendas fusionados.
*   **Volumen de Ventas e Ingresos (General):** Las cuatro tiendas exhiben volúmenes de ventas muy similares (2358-2359 artículos). Sin embargo, en términos de ingresos totales (precio del producto), la Tienda 1 es la que más gana con $1,150,880,400 COP ($297,769.83 USD), mientras que la Tienda 4 es la que menos gana con $1,038,375,700 COP ($268,661.24 USD).
*   **Métodos de Pago:** "Tarjeta de crédito" es el método de pago más preferido en todas las tiendas, seguido de "Nequi", siendo "Tarjeta de débito" el menos utilizado. Esta tendencia es uniforme.
*   **Rendimiento por Categoría de Producto:**
    *   **Categorías Principales:** "Electrónicos" y "Electrodomésticos" generan consistentemente la mayor parte de los ingresos para todas las tiendas (por ejemplo, Tienda 1: Electrónicos $429,493,500 COP/$111,123.80 USD, Electrodomésticos $363,685,200 COP/$94,097.08 USD).
    *   **Categorías Bajas:** "Libros" y "Artículos para el hogar" son consistentemente las categorías con menores ingresos en todas las tiendas (por ejemplo, Tienda 1: Libros $8,784,900 COP/$2,272.94 USD, Artículos para el hogar $12,698,400 COP/$3,285.49 USD).
*   **Calificaciones de las Tiendas:** Todas las tiendas mantienen calificaciones promedio de clientes consistentemente altas, rondando los 4.0 sobre 5, lo que indica una satisfacción uniforme del cliente.
*   **Distribución Geográfica:** Los principales centros urbanos —Bogotá (3943 compras), Medellín (2314 compras) y Cali (1007 compras)— representan la gran mayoría de las ventas en todas las tiendas y para las categorías de productos más vendidas, sin que ninguna tienda domine una ciudad específica.
*   **Costos de Envío:** Los costos de envío promedio son muy consistentes en todas las tiendas, aproximadamente $23,459.46 - $26,018.61 COP (aproximadamente $6-7 USD), lo que sugiere un enfoque logístico estandarizado.
*   **Rendimiento del Vendedor (Caso de Estudio Izabela de León):** Izabela de León es una vendedora con un alto volumen de ventas (por ejemplo, 197 artículos en la Tienda 1, 195 en la Tienda 2) pero también se clasifica constantemente como la que tiene la peor calificación promedio en la Tienda 1 (3.75) y la Tienda 3 (3.73). En la clasificación general de rendimiento (basada en el porcentaje de calificaciones altas frente a bajas), Izabela de León está en la parte inferior con una puntuación de 38.17, principalmente debido a un alto porcentaje de calificaciones por debajo de 3 (12.50%). Por el contrario, Ángel Rodríguez es el vendedor con mejor rendimiento con una puntuación de 58.18.
*   **Tendencias de Ingresos en Series de Tiempo (2020-2023):**
    *   La Tienda 4, a pesar de un fuerte comienzo en 2020 (segundo ingreso más alto), muestra una notable y consistente tendencia a la baja en los ingresos mensuales y anuales relativos.
    *   Para 2022 y los datos parciales de 2023, la Tienda 4 registra constantemente los ingresos mensuales más bajos, a menudo sin alcanzar el umbral de 30 millones de COP/mes mantenido por otras tiendas.

### Conclusiones y Próximos Pasos

*   **Revisión Estratégica del Portafolio de Productos:** Dado que "Libros" y "Artículos para el hogar" tienen un rendimiento consistentemente bajo, se justifica una revisión detallada de estas categorías en todas las tiendas para determinar si deben descontinuarse o si su estrategia debe redefinirse para mejorar la rentabilidad.
*   **Gestión del Rendimiento del Vendedor:** Abordar la disparidad de rendimiento observada en vendedores de alto volumen como Izabela de León. Implementar planes de capacitación o mejora del rendimiento dirigidos a los vendedores con un alto volumen de ventas pero baja satisfacción del cliente, ya que su impacto negativo en la experiencia del cliente podría superar sus contribuciones de ventas.

In [20]:
#Calculamos el total de ventas y calificación para cada vendedor a trvés de todas las tiendas y sacamos su rendimiento
vendor_performance = tiendas.groupby('Vendedor').agg(
    Total_Sales=('Producto', 'count'),
    Numero_Calificaciones_Superiores_a_4=('Calificación', lambda x: (x > 4).sum()),
    Numero_Calificaciones_Inferiores_a_3=('Calificación', lambda x: (x < 3).sum())
).reset_index()

# Calculamos los porcentajes
vendor_performance['% Calificaciones Superiores a 4'] = (vendor_performance['Numero_Calificaciones_Superiores_a_4'] / vendor_performance['Total_Sales']) * 100
vendor_performance['% Calificaciones Inferiores a 3'] = (vendor_performance['Numero_Calificaciones_Inferiores_a_3'] / vendor_performance['Total_Sales']) * 100
vendor_performance['% Otras Calificaciones'] = 100 - (vendor_performance['% Calificaciones Superiores a 4'] + vendor_performance['% Calificaciones Inferiores a 3'])

# Calculamos la calificación promedio para cada vendedor en cada tienda
vendor_store_ratings = tiendas.groupby(['Vendedor', 'Tienda'])['Calificación'].mean().reset_index(name='Average_Rating')

# Encontramos la tienda en donde cada vendedor tuvo mejor calificación
best_store_rating_idx = vendor_store_ratings.groupby('Vendedor')['Average_Rating'].idxmax()
best_store_rating = vendor_store_ratings.loc[best_store_rating_idx][['Vendedor', 'Tienda', 'Average_Rating']].rename(
    columns={'Tienda': 'Best_Performance_Store', 'Average_Rating': 'Best_Average_Rating'}
)

# Encontramos la tienda donde cada vendedor obtuvo la peor calificacion
worst_store_rating_idx = vendor_store_ratings.groupby('Vendedor')['Average_Rating'].idxmin()
worst_store_rating = vendor_store_ratings.loc[worst_store_rating_idx][['Vendedor', 'Tienda', 'Average_Rating']].rename(
    columns={'Tienda': 'Worst_Performance_Store', 'Average_Rating': 'Worst_Average_Rating'}
)

# Combinamos toda la información mejores y peores
best_worst_store_rating = best_store_rating.merge(worst_store_rating, on='Vendedor')


# Combinamos los datos de redimiento con los datos mejores/peores
final_vendor_ranking = vendor_performance.merge(best_worst_store_rating, on='Vendedor', how='left')

# Rankeamos los vendedores basandonos en el rendimiendo (tenidendo en cuenta diferencia entre procentajes positivos y negativos)
final_vendor_ranking['Performance_Score'] = final_vendor_ranking['% Calificaciones Superiores a 4'] - final_vendor_ranking['% Calificaciones Inferiores a 3']

# Organizamos de forma descendente por el score de rendimiento
final_vendor_ranking = final_vendor_ranking.sort_values(by='Performance_Score', ascending=False)

# Formateamos columnas a 2 decimales
final_vendor_ranking['% Calificaciones Superiores a 4'] = final_vendor_ranking['% Calificaciones Superiores a 4'].map('{:.2f}'.format)
final_vendor_ranking['% Calificaciones Inferiores a 3'] = final_vendor_ranking['% Calificaciones Inferiores a 3'].map('{:.2f}'.format)
final_vendor_ranking['% Otras Calificaciones'] = final_vendor_ranking['% Otras Calificaciones'].map('{:.2f}'.format)


# Mostramos con las calificaciones y procentajes de rendimiento
print("Vendor Ranking by Performance:")
display(final_vendor_ranking[['Vendedor', 'Total_Sales', 'Numero_Calificaciones_Superiores_a_4', 'Numero_Calificaciones_Inferiores_a_3', '% Calificaciones Superiores a 4', '% Calificaciones Inferiores a 3', '% Otras Calificaciones', 'Best_Performance_Store', 'Worst_Performance_Store', 'Performance_Score']])

Vendor Ranking by Performance:


Vendedor  Total_Sales  Numero_Calificaciones_Superiores_a_4  \
13  Ángel Rodriguez          684                                   407   
11    Rafael Acosta          625                                   366   
3     Felipe Santos          701                                   408   
9   Mariana Herrera          656                                   366   
0   Beatriz Morales          650                                   375   
2     Camila Rivera          667                                   372   
5    Juan Fernandez          694                                   389   
10      Pedro Gomez          655                                   369   
1    Blanca Ramirez          679                                   389   
12   Santiago Silva          650                                   352   
8     Maria Alfonso          679                                   364   
7     Lucas Olivera          674                                   355   
6     Juliana Costa          677                                   361   
4   Izabela de León          744                                   377   

    Numero_Calificaciones_Inferiores_a_3 % Calificaciones Superiores a 4  \
13                                    95                           59.50   
11                                    95                           58.56   
3                                    108                           58.20   
9                                     93                           55.79   
0                                    107                           57.69   
2                                    100                           55.77   
5                                    111                           56.05   
10                                   108                           56.34   
1                                    120                           57.29   
12                                   101                           54.15   
8                                    116                           53.61   
7                                    110                           52.67   
6                                    127                           53.32   
4                                    151                           50.67   

   % Calificaciones Inferiores a 3 % Otras Calificaciones  \
13                           13.89                  26.61   
11                           15.20                  26.24   
3                            15.41                  26.39   
9                            14.18                  30.03   
0                            16.46                  25.85   
2                            14.99                  29.24   
5                            15.99                  27.95   
10                           16.49                  27.18   
1                            17.67                  25.04   
12                           15.54                  30.31   
8                            17.08                  29.31   
7                            16.32                  31.01   
6                            18.76                  27.92   
4                            20.30                  29.03   

   Best_Performance_Store Worst_Performance_Store  Performance_Score  
13               Tienda 1                Tienda 4          45.614035  
11               Tienda 3                Tienda 4          43.360000  
3                Tienda 3                Tienda 4          42.796006  
9                Tienda 2                Tienda 1          41.615854  
0                Tienda 3                Tienda 2          41.230769  
2                Tienda 2                Tienda 4          40.779610  
5                Tienda 3                Tienda 1          40.057637  
10               Tienda 3                Tienda 4          39.847328  
1                Tienda 4                Tienda 2          39.617084  
12               Tienda 4                Tienda 2          38.615385  
8                Tienda 2                Tienda 1          36.

In [21]:
import plotly.express as px

# Sort the final_vendor_ranking DataFrame by Performance_Score en order descendente
ranked_vendors = final_vendor_ranking.sort_values(by='Performance_Score', ascending=False)

# Creamos un gráfico de barras para la calificación de rendimiento utilizando Plotly Express
fig = px.bar(ranked_vendors,
             x='Performance_Score',
             y='Vendedor',
             orientation='h', # Orientación horizontal para mejor lectura de nombres largos
             title='Ranking de Rendimiento de Vendedores',
             labels={'Performance_Score': 'Performance Score (% Calificaciones Superiores a 4 - % Calificaciones Inferiores a 3)'},
             hover_data={'Total_Sales': True,
                         'Numero_Calificaciones_Superiores_a_4': True,
                         'Numero_Calificaciones_Inferiores_a_3': True,
                         '% Calificaciones Superiores a 4': True,
                         '% Calificaciones Inferiores a 3': True,
                         '% Otras Calificaciones': True,
                         'Best_Performance_Store': True,
                         'Worst_Performance_Store': True,
                         'Performance_Score': ':.2f'}) # Formato para el hover del performance score

# Mejoramos la visualización del gráfico
fig.update_layout(yaxis={'categoryorder':'total ascending'}) # Ordenar vendedores por performance score
fig.show()

#### **Gráfico de columna-línea de rendimiento de vendedores**

# 9. Graficas complementarias

#### **Cantidad de artículos vendidos por categoría por tienda**

In [22]:
# Agrupamos los datos por tienda y categoría para contar la cantidad de productos vendidos
ventas_por_categoria = tiendas.groupby(["Tienda", "Categoría del Producto"]).size().reset_index(name="Cantidad de Artículos Vendidos")

# Creamos el gráfico de barras utilizando Plotly Express
fig = px.bar(ventas_por_categoria,
             x="Categoría del Producto",
             y="Cantidad de Artículos Vendidos",
             color="Tienda", # Diferenciamos las barras por tienda
             barmode="group", # Agrupamos las barras por categoría
             title="Cantidad de Artículos Vendidos por Categoría y Tienda")

# Mejoramos la visualización del gráfico
fig.update_layout(xaxis={'categoryorder':'total descending'}) # Ordenamos las categorías por la cantidad total vendida
fig.show()

In [23]:
import plotly.express as px
import pandas as pd

# Extraemos el año y el mes de la fecha de compra
tiendas['Año'] = tiendas['Fecha de Compra'].dt.year
tiendas['Mes_num'] = tiendas['Fecha de Compra'].dt.month # Extraemos el número del mes

# Creamos un rango de fechas completo para 2020, 2021 y 2022 (todos los meses)
full_date_range_2020_2022 = pd.date_range(start='2020-01-01', end='2022-12-31', freq='ME')

# Preparamos un DataFrame con todas las combinaciones de tienda y mes para 2020-2022
all_months_stores_2020_2022 = pd.MultiIndex.from_product([
    full_date_range_2020_2022,
    tiendas['Tienda'].unique()
], names=['Fecha de Compra', 'Tienda']).to_frame(index=False)

# Agregamos el año y el número del mes a este dataframe
all_months_stores_2020_2022['Año'] = all_months_stores_2020_2022['Fecha de Compra'].dt.year
all_months_stores_2020_2022['Mes_num'] = all_months_stores_2020_2022['Fecha de Compra'].dt.month


# Agrupamos los datos por año, mes y tienda para calcular los ingresos totales
# Para 2020, 2021, 2022, agrupamos y luego nos unimos con el rango completo para llenar los meses faltantes
ingresos_por_año_mes_tienda_2020_2022 = tiendas[tiendas['Año'].isin([2020, 2021, 2022])].groupby([
    'Año',
    pd.Grouper(key='Fecha de Compra', freq='ME'),
    'Tienda'
])['Precio'].sum().reset_index(name='Ingresos Totales')

# Agregamos el número del mes a este dataframe para la unión
ingresos_por_año_mes_tienda_2020_2022['Mes_num'] = ingresos_por_año_mes_tienda_2020_2022['Fecha de Compra'].dt.month


# Nos unimos con el rango completo usando 'Año', 'Tienda', y 'Mes_num'
ingresos_completos_2020_2022 = all_months_stores_2020_2022.merge(
    ingresos_por_año_mes_tienda_2020_2022[['Año', 'Tienda', 'Mes_num', 'Ingresos Totales']],
    on=['Año', 'Tienda', 'Mes_num'],
    how='left'
).fillna(0)


# Para 2023, solo agrupamos los datos existentes (no llenamos meses sin ventas)
ingresos_por_año_mes_tienda_2023 = tiendas[tiendas['Año'] == 2023].groupby([
    'Año',
    pd.Grouper(key='Fecha de Compra', freq='ME'),
    'Tienda'
])['Precio'].sum().reset_index(name='Ingresos Totales')

# Agregamos el número del mes a este dataframe
ingresos_por_año_mes_tienda_2023['Mes_num'] = ingresos_por_año_mes_tienda_2023['Fecha de Compra'].dt.month


# Concatenamos los dataframes de 2020-2022 (completos) y 2023 (solo con datos)
ingresos_combinados = pd.concat([ingresos_completos_2020_2022, ingresos_por_año_mes_tienda_2023[['Año', 'Tienda', 'Mes_num', 'Ingresos Totales']]], ignore_index=True)


# Calculamos el ingreso total por tienda para ordenar las líneas (en todo el período)
ingresos_totales_por_tienda_combinado = ingresos_combinados.groupby('Tienda')['Ingresos Totales'].sum().sort_values(ascending=False)

# Creamos una lista de nombres abreviados de los meses
month_names_abbr = ["Ene", "Feb", "Mar", "Abr", "May", "Jun", "Jul", "Ago", "Sep", "Oct", "Nov", "Dic"]

# Creamos el gráfico de línea de tiempo con facetas por año, usando Mes_num en el eje X
fig = px.line(ingresos_combinados,
              x='Mes_num', # Usamos el número del mes en el eje X
              y='Ingresos Totales',
              color='Tienda', # Diferenciar líneas por tienda
              facet_col='Año', # Separar por año
              facet_col_wrap=2, # Envolver las columnas después de 2 años
              title='Ingresos Mensuales por Tienda Separados por Año (COP)',
              category_orders={"Tienda": ingresos_totales_por_tienda_combinado.index.tolist(),
                               "Año": sorted(ingresos_combinados['Año'].unique())}) # Ordenamos las tiendas y años

# Ajustamos el diseño para una mejor visualización y formato de moneda
fig.update_layout(
    xaxis=dict(
        tickvals=list(range(1, 13)), # Aseguramos ticks para todos los meses (1-12)
        ticktext=month_names_abbr, # Mostramos las iniciales de los meses como texto de tick
        rangeslider_visible=False, # No necesitamos rangeslider en facetas por año
    ),
    yaxis=dict(
        tickformat=",.0f", # Formato para pesos colombianos sin decimales
        hoverformat=",.0f" # Formato para hover
    ),
    hovermode="x unified",
    legend=dict(
        traceorder="normal" # Aseguramos el orden normal de la leyenda
    )
)

fig.show()

In [24]:
# Verificamos si hay registros para cada mes de cada año
ventas_por_mes_año = tiendas.groupby(pd.Grouper(key='Fecha de Compra', freq='MS')).size().reset_index(name='Cantidad de Ventas')

print("Cantidad de ventas por mes y año:")
display(ventas_por_mes_año)

# Para verificar si hay meses faltantes, podemos crear un rango completo de fechas
# desde la primera venta hasta la última y compararlo con las fechas existentes.
rango_completo_fechas = pd.date_range(start=tiendas['Fecha de Compra'].min(),
                                     end=tiendas['Fecha de Compra'].max(),
                                     freq='MS')

fechas_existentes = ventas_por_mes_año['Fecha de Compra']

meses_faltantes = rango_completo_fechas[~rango_completo_fechas.isin(fechas_existentes)]

if meses_faltantes.empty:
    print("\nHay registros de ventas para cada mes desde la primera hasta la última fecha.")
else:
    print("\nMeses faltantes en los registros de ventas:")
    display(meses_faltantes)

Cantidad de ventas por mes y año:


Fecha de Compra  Cantidad de Ventas
0       2020-01-01                 267
1       2020-02-01                 226
2       2020-03-01                 284
3       2020-04-01                 243
4       2020-05-01                 239
5       2020-06-01                 211
6       2020-07-01                 251
7       2020-08-01                 224
8       2020-09-01                 247
9       2020-10-01                 192
10      2020-11-01                 235
11      2020-12-01                 242
12      2021-01-01                 252
13      2021-02-01                 236
14      2021-03-01                 242
15      2021-04-01                 244
16      2021-05-01                 227
17      2021-06-01                 224
18      2021-07-01                 261
19      2021-08-01                 257
20      2021-09-01                 248
21      2021-10-01                 250
22      2021-11-01                 267
23      2021-12-01                 274
24      2022-01-01                 250
25      2022-02-01                 222
26      2022-03-01                 248
27      2022-04-01                 247
28      2022-05-01                 227
29      2022-06-01                 225
30      2022-07-01                 222
31      2022-08-01                 249
32      2022-09-01                 260
33      2022-10-01                 220
34      2022-11-01                 250
35      2022-12-01                 253
36      2023-01-01                 216
37      2023-02-01                 236
38      2023-03-01                 267


Hay registros de ventas para cada mes desde la primera hasta la última fecha.


#### **Ingresos Mensuales por Tienda - Año 2020**

In [25]:
import plotly.express as px
import pandas as pd

# Filtramos los datos para el año 2020
tiendas_2020 = tiendas[tiendas['Fecha de Compra'].dt.year == 2020].copy()

# Creamos un rango de fechas completo para 2020 (todos los meses) y ajustamos a fin de mes
full_date_range_2020 = pd.date_range(start='2020-01-01', end='2020-12-31', freq='ME') # Cambiado de 'MS' a 'ME'

# Preparamos un DataFrame con todas las combinaciones de tienda y mes para 2020
all_months_stores_2020 = pd.MultiIndex.from_product([
    full_date_range_2020,
    tiendas_2020['Tienda'].unique()
], names=['Fecha de Compra', 'Tienda']).to_frame(index=False)


# Agrupamos los datos de 2020 por mes y tienda y calculamos los ingresos
ingresos_por_mes_tienda_2020 = tiendas_2020.groupby([
    pd.Grouper(key='Fecha de Compra', freq='ME'),
    'Tienda'
])['Precio'].sum().reset_index(name='Ingresos Totales')

# Nos unimos con el rango completo para asegurar todos los meses y llenamos los valores faltantes con 0
ingresos_completos_2020 = all_months_stores_2020.merge(
    ingresos_por_mes_tienda_2020,
    on=['Fecha de Compra', 'Tienda'],
    how='left'
).fillna(0)

# Renombramos la columna de fecha para mayor claridad en el gráfico
ingresos_completos_2020.rename(columns={'Fecha de Compra': 'Mes'}, inplace=True)

# VERIFICACIÓN: Mostrar los primeros registros del dataframe antes de graficar
print("\nDatos de ingresos completos para 2020 antes de graficar:")
display(ingresos_completos_2020.head())
print("\nInformación del dataframe de ingresos completos para 2020:")
ingresos_completos_2020.info()


# Calculamos el ingreso total por tienda para ordenar las líneas
ingresos_totales_por_tienda_2020_orden = ingresos_completos_2020.groupby('Tienda')['Ingresos Totales'].sum().sort_values(ascending=False)


# Creamos el gráfico de línea de tiempo para 2020
fig = px.line(ingresos_completos_2020,
              x='Mes',
              y='Ingresos Totales',
              color='Tienda', # Diferenciar líneas por tienda
              title='Ingresos Mensuales por Tienda en 2020 (COP)',
              category_orders={"Tienda": ingresos_totales_por_tienda_2020_orden.index.tolist()}) # Ordenamos las tiendas

# Ajustamos el diseño para formato de moneda y rango del eje X
fig.update_layout(
    xaxis=dict(
        tickformat="%Y-%m",
        range=[full_date_range_2020.min(), full_date_range_2020.max()], # Aseguramos el rango completo del año
    ),
    yaxis=dict(
        tickformat=",.0f",
        hoverformat=",.0f"
    ),
    hovermode="x unified",
    legend=dict(
        traceorder="normal"
    )
)

fig.show()


Datos de ingresos completos para 2020 antes de graficar:


Mes    Tienda  Ingresos Totales
0 2020-01-31  Tienda 1        33084300.0
1 2020-01-31  Tienda 2        31129500.0
2 2020-01-31  Tienda 3        23782900.0
3 2020-01-31  Tienda 4        19298700.0
4 2020-02-29  Tienda 1        40054000.0


Información del dataframe de ingresos completos para 2020:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48 entries, 0 to 47
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   Mes               48 non-null     datetime64[ns]
 1   Tienda            48 non-null     object        
 2   Ingresos Totales  48 non-null     float64       
dtypes: datetime64[ns](1), float64(1), object(1)
memory usage: 1.3+ KB


#### **Ingresos Mensuales por Tienda - Año 2021**

In [26]:
import plotly.express as px
import pandas as pd

# Filtramos los datos para el año 2021
tiendas_2021 = tiendas[tiendas['Fecha de Compra'].dt.year == 2021].copy()

# Creamos un rango de fechas completo para 2021 (todos los meses) y ajustamos a fin de mes
full_date_range_2021 = pd.date_range(start='2021-01-01', end='2021-12-31', freq='ME')

# Preparamos un DataFrame con todas las combinaciones de tienda y mes para 2021
all_months_stores_2021 = pd.MultiIndex.from_product([
    full_date_range_2021,
    tiendas_2021['Tienda'].unique()
], names=['Fecha de Compra', 'Tienda']).to_frame(index=False)

# Agrupamos los datos de 2021 por mes y tienda y calculamos los ingresos
ingresos_por_mes_tienda_2021 = tiendas_2021.groupby([
    pd.Grouper(key='Fecha de Compra', freq='ME'),
    'Tienda'
])['Precio'].sum().reset_index(name='Ingresos Totales')

# Nos unimos con el rango completo para asegurar todos los meses y llenamos los valores faltantes con 0
ingresos_completos_2021 = all_months_stores_2021.merge(
    ingresos_por_mes_tienda_2021,
    on=['Fecha de Compra', 'Tienda'],
    how='left'
).fillna(0)

# Renombramos la columna de fecha para mayor claridad en el gráfico
ingresos_completos_2021.rename(columns={'Fecha de Compra': 'Mes'}, inplace=True)

# Calculamos el ingreso total por tienda para ordenar las líneas
ingresos_totales_por_tienda_2021_orden = ingresos_completos_2021.groupby('Tienda')['Ingresos Totales'].sum().sort_values(ascending=False)


# Creamos el gráfico de línea de tiempo para 2021
fig = px.line(ingresos_completos_2021,
              x='Mes',
              y='Ingresos Totales',
              color='Tienda', # Diferenciar líneas por tienda
              title='Ingresos Mensuales por Tienda en 2021 (COP)',
              category_orders={"Tienda": ingresos_totales_por_tienda_2021_orden.index.tolist()}) # Ordenamos las tiendas

# Ajustamos el diseño para formato de moneda y rango del eje X
fig.update_layout(
    xaxis=dict(
        tickformat="%Y-%m",
        range=[full_date_range_2021.min(), full_date_range_2021.max()], # Aseguramos el rango completo del año
    ),
    yaxis=dict(
        tickformat=",.0f",
        hoverformat=",.0f"
    ),
    hovermode="x unified",
    legend=dict(
        traceorder="normal"
    )
)

fig.show()

#### **Ingresos Mensuales por Tienda - Año 2022**

In [27]:
import plotly.express as px
import pandas as pd

# Filtramos los datos para el año 2022
tiendas_2022 = tiendas[tiendas['Fecha de Compra'].dt.year == 2022].copy()

# Creamos un rango de fechas completo para 2022 (todos los meses) y ajustamos a fin de mes
full_date_range_2022 = pd.date_range(start='2022-01-01', end='2022-12-31', freq='ME')

# Preparamos un DataFrame con todas las combinaciones de tienda y mes para 2022
all_months_stores_2022 = pd.MultiIndex.from_product([
    full_date_range_2022,
    tiendas_2022['Tienda'].unique()
], names=['Fecha de Compra', 'Tienda']).to_frame(index=False)

# Agrupamos los datos de 2022 por mes y tienda y calculamos los ingresos
ingresos_por_mes_tienda_2022 = tiendas_2022.groupby([
    pd.Grouper(key='Fecha de Compra', freq='ME'),
    'Tienda'
])['Precio'].sum().reset_index(name='Ingresos Totales')

# Nos unimos con el rango completo para asegurar todos los meses y llenamos los valores faltantes con 0
ingresos_completos_2022 = all_months_stores_2022.merge(
    ingresos_por_mes_tienda_2022,
    on=['Fecha de Compra', 'Tienda'],
    how='left'
).fillna(0)

# Renombramos la columna de fecha para mayor claridad en el gráfico
ingresos_completos_2022.rename(columns={'Fecha de Compra': 'Mes'}, inplace=True)

# Calculamos el ingreso total por tienda para ordenar las líneas
ingresos_totales_por_tienda_2022_orden = ingresos_completos_2022.groupby('Tienda')['Ingresos Totales'].sum().sort_values(ascending=False)


# Creamos el gráfico de línea de tiempo para 2022
fig = px.line(ingresos_completos_2022,
              x='Mes',
              y='Ingresos Totales',
              color='Tienda', # Diferenciar líneas por tienda
              title='Ingresos Mensuales por Tienda en 2022 (COP)',
              category_orders={"Tienda": ingresos_totales_por_tienda_2022_orden.index.tolist()}) # Ordenamos las tiendas

# Ajustamos el diseño para formato de moneda y rango del eje X
fig.update_layout(
    xaxis=dict(
        tickformat="%Y-%m",
        range=[full_date_range_2022.min(), full_date_range_2022.max()], # Aseguramos el rango completo del año
    ),
    yaxis=dict(
        tickformat=",.0f",
        hoverformat=",.0f"
    ),
    hovermode="x unified",
    legend=dict(
        traceorder="normal"
    )
)

fig.show()

#### **Ingresos Mensuales por Tienda - Año 2023**

In [28]:
import plotly.express as px
import pandas as pd

# Filtramos los datos para el año 2023
tiendas_2023 = tiendas[tiendas['Fecha de Compra'].dt.year == 2023].copy()

# Agrupamos los datos de 2023 por mes y tienda y calculamos los ingresos
ingresos_por_mes_tienda_2023 = tiendas_2023.groupby([
    pd.Grouper(key='Fecha de Compra', freq='ME'),
    'Tienda'
])['Precio'].sum().reset_index(name='Ingresos Totales')

# Renombramos la columna de fecha para mayor claridad en el gráfico
ingresos_por_mes_tienda_2023.rename(columns={'Fecha de Compra': 'Mes'}, inplace=True)

# Calculamos el ingreso total por tienda para ordenar las líneas
ingresos_totales_por_tienda_2023_orden = ingresos_por_mes_tienda_2023.groupby('Tienda')['Ingresos Totales'].sum().sort_values(ascending=False)


# Creamos el gráfico de línea de tiempo para 2023
fig = px.line(ingresos_por_mes_tienda_2023,
              x='Mes',
              y='Ingresos Totales',
              color='Tienda', # Diferenciar líneas por tienda
              title='Ingresos Mensuales por Tienda en 2023 (COP)',
              category_orders={"Tienda": ingresos_totales_por_tienda_2023_orden.index.tolist()}) # Ordenamos las tiendas

# Ajustamos el diseño para formato de moneda
fig.update_layout(
    xaxis=dict(
        tickformat="%Y-%m",
    ),
    yaxis=dict(
        tickformat=",.0f",
        hoverformat=",.0f"
    ),
    hovermode="x unified",
    legend=dict(
        traceorder="normal"
    )
)

fig.show()

# 10. Análisis final para Mr.Joao

## 📈 Tendencias de Ingresos Mensuales por Tienda (2020-2023)

Basándose en los gráficos de series de tiempo para 'Ingresos Mensuales por Tienda Separados por Año (COP)', 'Ingresos Mensuales por Tienda en 2020 (COP)', 'Ingresos Mensuales por Tienda en 2021 (COP)', 'Ingresos Mensuales por Tienda en 2022 (COP)' e 'Ingresos Mensuales por Tienda en 2023 (COP)', se pueden extraer las siguientes tendencias y observaciones:

---

## 💰 Tendencias Generales de Ingresos Mensuales (2020-2023)

* **Estabilidad General y Fluctuaciones:** Todas las tiendas exhiben **fluctuaciones mensuales** en los ingresos, lo que sugiere estacionalidad o condiciones de mercado variables a lo largo del año. Sin embargo, a lo largo de los años, la mayoría de las tiendas mantienen una **línea de base** de generación de ingresos relativamente constante.
* **Patrones de Picos y Valles:** Si bien los meses específicos de picos y valles varían, no existe un mes universalmente consistente en todos los años y para todas las tiendas para observar los ingresos más altos o más bajos. Esto indica que los **eventos locales** o las **estrategias de venta específicas** podrían influir en el rendimiento mensual en lugar de las amplias tendencias estacionales.

---

## 📅 Observaciones Anuales

### 2020
* **Tienda 1** muestra un **rendimiento sólido**, a menudo liderando o estando entre las que más ingresos generan. Experimentó un pico significativo a principios de año (p. ej., febrero) y mantuvo ventas robustas, terminando el año con ingresos altos. Su ingreso total para 2020 fue el **más alto** entre todas las tiendas ($368,933,200 COP).

* **Tienda 4** también muestra fuertes picos tempranos en 2020, incluso superando a otras tiendas en algunos meses, y terminó el año con un buen ingreso total ($330,847,700 COP), ocupando el **segundo lugar**.
* **Tienda 2 y Tienda 3** generalmente siguen patrones similares a la Tienda 1, pero con volúmenes generales ligeramente menores, aunque también experimentaron sus propios máximos mensuales.

### 2021
* **Tienda 3** emergió como la **tienda con mejor rendimiento** en 2021 con el ingreso total más alto ($362,954,400 COP), mostrando ingresos generalmente estables y crecientes a lo largo del año.

* **Tienda 1 y Tienda 2** continuaron con un rendimiento sólido, con la Tienda 1 muy cerca de la Tienda 3 en ingresos totales (362,120,800 COP), y la Tienda 2($351,215,500 COP) también demostrando una actividad consistente.

* El rendimiento de la **Tienda 4**, si bien sigue siendo sustancial ($347,823,600 COP), comenzó a mostrar cierto **descenso relativo** en comparación con su posición de 2020, ocupando ahora el cuarto lugar en ingresos totales anuales.

### 2022
* **Tienda 2** lideró la generación de ingresos en 2022 con $358,226,500 COP, mostrando picos notables en meses como enero y mayo.

* **Tienda 3** siguió siendo un fuerte competidor con $350,438,300 COP, mostrando un rendimiento constante. La **Tienda 1** también tuvo un buen rendimiento ($316,565,600 COP).

* La **Tienda 4** experimentó una **caída más pronunciada** en 2022, manteniéndose constantemente en el extremo inferior de los ingresos mensuales en comparación con otras tiendas y terminando el año con el **ingreso total más bajo** ($302,221,500 COP). Luchó por alcanzar los altos picos mensuales observados en años anteriores para las otras tiendas.

### 2023 (Datos Parciales)
* La tendencia de 2022 continuó en los meses registrados de 2023. La **Tienda 1** parece haberse recuperado para ser la **tienda con mejor rendimiento**(103,260,800 COP), seguida por la **Tienda 2** ($86,434,900 COP).

* La **Tienda 3** ($62,919,400 COP) y especialmente la **Tienda 4** (57,482,900 COP) muestran ingresos mensuales **significativamente más bajos** en comparación con las otras tiendas para los meses disponibles, lo que sugiere una **tendencia descendente continuada** o dificultad en la recuperación para la Tienda 4.

---

## 📉 Enfoque en el Rendimiento de la Tienda 4

* **Tendencia Decreciente:** La Tienda 4, que comenzó fuerte en 2020, ha mostrado una **tendencia descendente notable** en sus ingresos mensuales y anuales relativos en comparación con las otras tiendas. Para 2022 y 2023, registra consistentemente los ingresos mensuales totales y los ingresos anuales **más bajos**.
* **Picos Más Bajos:** A diferencia de sus contrapartes que logran alcanzar picos de ingresos mensuales más altos incluso en medio de las fluctuaciones, los picos mensuales de la Tienda 4 en los últimos años son **considerablemente más bajos** que los de las otras tiendas.
* **Resiliencia Reducida:** Parece que la Tienda 4 tiene **menos resiliencia o adaptabilidad** para mantener su volumen de ventas y generación de ingresos a lo largo del tiempo en comparación con la Tienda 1, Tienda 2 y Tienda 3.



## 📊 Resumen de Análisis de Tiendas y Rendimiento

Este resumen presenta las conclusiones clave del análisis de ventas, ganancias, métodos de pago, rendimiento del personal y tendencias históricas de las cuatro tiendas.

---

## 1. Volumen de Ventas e Ingresos Totales por Tienda

* **Volumen de Artículos:** Las cuatro tiendas muestran un volumen de ventas en artículos muy similar, con un promedio de 2,359 artículos vendidos de diferentes categorías. Solo la Tienda 4 registra una ligera diferencia con 2,358 artículos.
* **Ingresos Totales (Precio + Envío):**
    * **Tienda 1** es la que genera más ingresos, con un valor reportado de **$297,769 USD**.
    * **Tienda 4** es la que genera menos ingresos, con un valor de **$268,661 USD**.

---

## 2. Métodos de Pago y Rentabilidad de Productos

### 💳 Métodos de Pago Preferidos
Los métodos de pago más utilizados y con mayor volumen de transacciones en todas las tiendas son la **tarjeta de crédito** y **Nequi**.

### 📈 Productos con Mayor Venta vs. Mayor Ganancia
| Categoría de Producto | Volumen de Venta | Margen de Ganancia | Observaciones Clave |
| :--- | :--- | :--- | :--- |
| **Muebles** | Alto (Mayor número de ventas que Electrodomésticos) | Inferior | A pesar del alto volumen, el margen de ganancia es bajo. |
| **Electrónicos** | Alto | **Alto** | Generan la mayor ganancia. |
| **Electrodomésticos** | Moderado | **Alto** | Generan la mayor ganancia. |
| **Artículos para el hogar** | Bajo | Bajo | **Se recomienda reevaluar su venta.** |
| **Libros** | Bajo | Bajo | **Se recomienda reevaluar su venta.** |

> **Nota:** Las categorías de 'Artículos para el hogar' y 'Libros' no superan la barrera de los $4,000 USD generados al mes, en comparación con las otras categorías.

---

## 3. Calificación de Tiendas y Rendimiento de Vendedores

* **Calificación General de Tiendas:** Las calificaciones de las 4 tiendas son muy similares, con un **promedio de 4.0** en satisfacción del cliente.
* **Rendimiento de Vendedores:**
    * Se evidencia la necesidad de reevaluar al personal.
    * Vendedores como **Izabela de León**, a pesar de ser quien más vende en todas las tiendas, presenta la **peor calificación y rendimiento** en comparación con sus demás compañeros. Se requieren posibles mejoras y capacitación.

---

## 4. Distribución Geográfica y Precio de Envío

* **Distribución Geográfica:** Los productos más rentables (Electrónicos y Electrodomésticos) se distribuyen de manera muy similar en las mismas ciudades principales: **Medellín, Bogotá y Cali**. No existe una tendencia de venta concentrada en una sola ciudad/región específica por tienda.
* **Costo de Envío:** El precio de envío promedio es de **$6 USD** para todas las tiendas, por lo que no se evidencia una diferencia considerable en este aspecto.

---

## 5. Tendencias de Ingresos Históricos (2020-2023)

* **Análisis Histórico:** Se evaluó el rendimiento de ganancias generadas en una línea de tiempo para 2020, 2021, 2022 y los registros parciales de 2023.
* **Tienda 4 - Tendencia Descendente:**
    * En 2020 y 2021, la Tienda 4 tuvo repuntes de ingresos máximos/mes de **40 millones de pesos colombianos (M/mes)**.
    * En el año 2022 y en lo registrado de 2023, la Tienda 4 **no logra llegar a la barrera de los 30 M/mes**.
* **Otras Tiendas - Estabilidad:** Las demás tiendas logran mantenerse e incluso **superar los 30 M/mes** en el transcurso de los meses de los últimos años.

---

## ✅ Recomendación Adicional para Registro de Datos

Se hace necesario un **registro de costos** de mantenimiento, nómina y gastos operativos de cada tienda para contrastar con las ganancias generadas y establecer una **ganancia neta real** que refleje el verdadero rendimiento de cada tienda.

### 📌 5. Conclusiones y Recomendaciones Finales

#### Conclusiones Clave sobre el Rendimiento de la Tienda y Decisiones

Con base en el análisis, se recomienda el **cierre potencial** de la Tienda 4. Si bien las cuatro tiendas muestran volúmenes de ventas similares y calificaciones de satisfacción del cliente (con un promedio de 4.0), la Tienda 4 exhibe una **tendencia a la baja en las ventas** y una **rentabilidad inferior** en los últimos años (2022-2023), al no alcanzar consistentemente la marca de $30 millones COP/mes en ingresos que otras tiendas mantienen o superan.

* Las categorías de productos con mayores ingresos (**Electrónica** y **Electrodomésticos**) se distribuyen de manera similar en todas las ciudades para todas las tiendas.
* Productos como 'Muebles' tienen un alto volumen de ventas pero **márgenes de ganancia más bajos** en comparación con Electrónica y Electrodomésticos.
* 'Artículos para el hogar' y 'Libros' son consistentemente las categorías **menos vendidas y menos rentables** en todas las tiendas, lo que sugiere la necesidad de reevaluar su presencia.
* El análisis destacó que los vendedores de alto volumen, como Izabela de León, pueden tener **calificaciones promedio más bajas**, lo que indica la necesidad de revisar el desempeño y la capacitación de los vendedores individuales.

---

#### Recomendaciones para Mejoras en el Registro de Datos

Para mejorar la toma de decisiones futuras, se recomiendan las siguientes mejoras en el registro de datos:

1.  **Diferenciar Canales de Venta**: Registrar claramente si las compras se realizan **en persona o en línea**. Esto ayudará a identificar las preferencias del cliente por cada canal y a optimizar la logística, lo que podría **reducir los costos de envío** al ubicar estratégicamente las tiendas más cerca de los compradores en línea.
2.  **Rastrear Costos Operacionales**: Implementar un sistema para registrar los **gastos operativos** de cada tienda, como mantenimiento y nómina. Esto permitirá calcular una **verdadera utilidad neta** por tienda, proporcionando una evaluación más precisa del rendimiento y la rentabilidad de cada tienda individual.

¿Necesitas que te ayude a formatear o traducir algún otro texto?

# 11. README.md

## 📈 Análisis de Rendimiento de Tiendas y Recomendaciones Estratégicas (2020-2023)

### 📝 Resumen del Notebook

Este *notebook* realiza un análisis en profundidad de los datos de ventas de cuatro tiendas distintas (`Tienda 1` a `Tienda 4`) para evaluar su rendimiento, identificar tendencias clave y proporcionar recomendaciones estratégicas a las partes interesadas. El objetivo principal es determinar qué tienda, si acaso, debe ser considerada para el cierre, basándose en diversas métricas financieras y operacionales.

### 📊 Conjunto de Datos Utilizado
El análisis utiliza datos de transacciones de ventas de cuatro archivos CSV separados (`tienda_1.csv`, `tienda_2.csv`, `tienda_3.csv`, `tienda_4.csv`), los cuales se consolidan en un único *Pandas DataFrame*. Este conjunto de datos exhaustivo incluye información detallada como:
- **Product (Producto):** Nombre del artículo vendido.
- **Categoría del Producto:** Categoría a la que pertenece el producto.
- **Precio:** Precio del producto.
- **Costo de envío:** Costo del envío.
- **Fecha de Compra:** Fecha en que se realizó la compra.
- **Vendedor:** Nombre del vendedor.
- **Lugar de Compra:** Ciudad donde se realizó la compra.
- **Calificación:** Puntuación del cliente para la transacción.
- **Método de pago:** Método de pago utilizado.
- **Cantidad de cuotas:** Número de cuotas del pago.
- **lat/lon:** Coordenadas geográficas de la ubicación de la compra.
- **Tienda:** Identificador de la tienda donde ocurrió la venta.

### 🚀 Resumen del Proceso Analítico
El *notebook* sigue un enfoque analítico estructurado:
1. **Importación y Preparación de Datos:** Carga de datos de las cuatro tiendas, fusión y realización de conversiones iniciales de tipos de datos (p. ej., convertir 'Fecha de Compra' a objetos `datetime`).
2. **Análisis Exploratorio de Datos (EDA):** Inspección inicial de la estructura del conjunto de datos, estadísticas básicas, valores únicos y una revisión exhaustiva de valores faltantes o duplicados para garantizar la calidad de los datos.
3. **Análisis de Facturación:** Evaluación del volumen de ventas, ingresos totales (precio + costo de envío) tanto en COP como en USD, y análisis de las preferencias de pago de los clientes en todas las tiendas.
4. **Ventas por Categoría:** Análisis del número de artículos vendidos y los ingresos totales generados por cada categoría de producto para cada tienda, incluyendo visualizaciones.
5. **Rendimiento de Producto/Categoría:** Identificación de las categorías de productos y productos individuales con mayores y menores ingresos para cada tienda, basándose en los ingresos totales.
6. **Evaluación del Rendimiento de la Tienda:** Evaluación de la calificación promedio de los clientes para cada tienda.
7. **Análisis Geográfico:** Examen del volumen de compras por ciudad y visualización de la distribución geográfica de las categorías de productos más vendidos en Colombia para cada tienda.
8. **Análisis del Costo de Envío:** Cálculo del costo de envío promedio por tienda.
9. **Análisis del Rendimiento del Vendedor:** Evaluación detallada del rendimiento individual de los vendedores basándose en el volumen de ventas, las calificaciones promedio y el porcentaje de calificaciones altas/bajas, identificando a los vendedores y tiendas con mejor y peor rendimiento para cada uno.
10. **Análisis de Ingresos de Series de Tiempo:** Visualización de las tendencias de ingresos mensuales para cada tienda a lo largo de varios años (2020-2023) para comprender los patrones estacionales y el rendimiento a largo plazo.
11. **Recomendaciones:** Basado en el análisis exhaustivo, proporcionar una conclusión final sobre el rendimiento de la tienda y recomendaciones accionables para Mr. Joao, incluyendo sugerencias para mejorar la recopilación de datos.

---

## 📊 Conclusiones Clave del Rendimiento de Tiendas

### 1. Volumen de Ventas e Ingresos Totales

| Métrica | Tienda 1 | Tienda 4 | Comentario General |
| :--- | :--- | :--- | :--- |
| Volumen de Artículos | 2,359 | 2,358 | Muy similar entre las 4 tiendas.|
| Ingresos Totales (USD) | **297,769 USD** | **268,661 USD** | Tienda 1 lidera; Tienda 4 es la de menor ingreso.|

Tienda 2 y 3 están por encima de los ingresos de Tienda 4.

### 2. Tendencias de Ingresos Históricos (2020-2023)

La métrica más crítica es la **tendencia descendente de la Tienda 4**.

* **Periodo 2020-2021:** Tienda 4 alcanzó picos máximos de **40 Millones COP/mes**.
* **Periodo 2022-2023:** Tienda 4 **no logra superar la barrera de los 30 Millones COP/mes**, mientras que las demás tiendas mantienen o superan consistentemente este umbral.

### 3. Rendimiento de Categorías y Métodos de Pago

* **Métodos de Pago Preferidos:** **Tarjeta de crédito** y **Nequi** son los más utilizados en todas las tiendas.
* **Categorías Más Rentables:** **Electrónicos** y **Electrodomésticos** generan las mayores ganancias, a pesar de que 'Muebles' tiene un alto volumen de ventas con un margen de ganancia inferior.
* **Categorías de Baja Rentabilidad:** **Artículos para el hogar** y **Libros** son las menos vendidas y menos rentables (no superan los $4,000 USD/mes).
* **Distribución Geográfica:** Los productos más rentables se distribuyen de manera uniforme en **Medellín, Bogotá y Cali**.

### 4. Calificación y Rendimiento del Personal

* **Calificación de Tiendas:** El promedio de satisfacción del cliente es similar en todas las tiendas, con **4.0**.
* **Rendimiento de Vendedores:** Se identificó a **Izabela de León** como la vendedora con el mayor volumen de ventas, pero con la **peor calificación y rendimiento** en comparación con sus compañeros, lo que indica un problema de calidad que requiere atención.

---

## 🔍 Análisis Detallado de Rendimiento de Tiendas y Estrategia Operativa (2020-2023)

Este análisis amplía las conclusiones clave sobre el desempeño financiero, la eficiencia operativa y las tendencias históricas de las cuatro tiendas, ofreciendo una base más sólida para la toma de decisiones estratégicas.

---

### 1. 💰 Desempeño Financiero y de Volumen

#### 1.1. Volumen de Ventas e Ingresos Totales
Las tiendas muestran una notable **homogeneidad en la capacidad de mover inventario**, con un volumen de artículos vendidos que apenas varía entre ellas.

* **Volumen de Artículos:** 2,359 artículos en promedio. La Tienda 4 (2,358) no difiere significativamente en volumen de transacciones.
* **Disparidad de Ingresos:** A pesar del volumen similar, existe una brecha de ingresos notable:
    * **Tienda 1 (Líder):** $297,769 USD en ingresos totales.
    * **Tienda 4 (Rezago):** $268,661 USD en ingresos totales.

> **Implicación:** La Tienda 4 está vendiendo la misma cantidad de artículos, pero su cesta promedio o la mezcla de productos vendidos es menos valiosa que la de sus competidores, lo que ya sugiere problemas de mezcla de inventario o precios.

#### 1.2. Rentabilidad por Categoría y Estrategia de Inventario

El análisis de categorías revela una **clara necesidad de optimización de inventario**:

| Categoría de Producto | Estrategia de Venta | Recomendación de Estrategia |
| :--- | :--- | :--- |
| **Electrónicos y Electrodomésticos** | Alto Margen de Ganancia | **Foco Prioritario:** Maximizar el inventario y la estrategia de *marketing* en estas categorías. |
| **Muebles** | Alto Volumen, Bajo Margen | **Revisión de Margen:** Evaluar costos de adquisición, almacenamiento y logística. El alto volumen no justifica la baja rentabilidad. |
| **Artículos para el hogar y Libros** | Bajo Volumen y Baja Ganancia | **Descontinuación/Reducción:** Considerar eliminar estas líneas en las tiendas que no superan los $4,000 USD/mes, liberando espacio y recursos. |

---

### 2. 📉 Crisis de la Tienda 4: Tendencia Histórica

El principal riesgo para la empresa se centra en la sosteniblidad de la Tienda 4, como lo demuestra el análisis de series de tiempo.

* **Estabilidad General (2020-2023):** Tienda 1, Tienda 2 y Tienda 3 han demostrado **resiliencia**, manteniendo o superando consistentemente el umbral de **30 Millones COP/mes**.
* **Declive de la Tienda 4:** La tienda tuvo buenos rendimientos en 2020-2021 (picos de 40 Millones COP/mes), pero a partir de 2022 y en los registros de 2023, ha **fallado consistentemente en alcanzar los 30 Millones COP/mes**.

> **Conclusión Financiera:** La Tienda 4 ya no opera al nivel de sus pares y está en una trayectoria de declive sostenido. Su volumen de ventas ya no se traduce en ingresos competitivos.

---

### 3. 👥 Rendimiento Operacional y de Personal

#### 3.1. Calidad de Servicio y Capacitación
* **Calificación General:** El promedio de **4.0** es aceptable, pero indica que existe margen de mejora en la satisfacción del cliente.
* **El Problema Izabela de León:** El caso de esta vendedora ilustra una desconexión entre la eficiencia en la venta y la calidad del servicio. Su alto volumen de ventas con la **peor calificación promedio** sugiere:
    * **Necesidad de Intervención:** Urgente capacitación enfocada en servicio al cliente.
    * **Riesgo de Marca:** Su bajo rendimiento puede estar dañando la reputación de la marca a pesar de generar altos ingresos.

#### 3.2. Geografía y Costos de Envío
* **Estrategia Geográfica:** Los puntos calientes de venta (**Medellín, Bogotá y Cali**) son consistentes para los productos más rentables, lo que sugiere que la distribución actual de las tiendas es adecuada.
* **Costo de Envío:** El costo promedio de **$6 USD** es uniforme, eliminándolo como un factor de diferenciación o problema logístico entre las tiendas.

---

### 🎯 Conclusión Final y Recomendaciones
Es fundamental complementar el análisis con datos de costos para calcular la verdadera **Utilidad Neta**.

#### Conclusiones Clave

1.  **Recomendación de Cierre (Tienda 4):** Dada su persistente tendencia a la baja en ventas y su inferior rentabilidad en 2022-2023, se recomienda el **cierre potencial** de la Tienda 4.
2.  **Reevaluación de Stock:** Se debe considerar **replantear la venta** de 'Artículos para el hogar' y 'Libros' en todas las tiendas debido a su baja rentabilidad.
3.  **Acción de Personal:** Es crucial **revisar el desempeño y capacitación** del personal, especialmente de vendedores de alto volumen con bajas calificaciones (ej. Izabela de León).
4.   **Registro de Costos Operacionales:** Implementar un sistema de contabilidad por centro de costo para capturar gastos como **mantenimiento, nómina y gastos fijos** de cada tienda. Esto permitirá calcular la:

    **Utilidad Neta = Ingresos Totales Costo de Bienes Vendidos - Gastos Operativos**

#### 📝Recomendaciones para Mejoras en el Registro de Datos

Para una toma de decisiones más precisa en el futuro, se requiere la implementación de las siguientes mejoras en la captura de datos:

1.  **Diferenciación de Canales de Venta:** Es vital registrar **'En Persona' vs. 'Online'** para cada transacción. Esta información permitirá:
    * **Optimizar Logística:** Tomar decisiones sobre dónde ubicar inventario o centros de distribución para reducir los $6 USD de costo de envío promedio.
    * **Entender al Cliente:** Adaptar promociones y la experiencia de compra a la preferencia de cada canal.
2.  **Rastrear Costos Operacionales:** Implementar un sistema para registrar **gastos operativos** (mantenimiento, nómina, etc.) por tienda. Esto es esencial para calcular una **verdadera utilidad neta** y el rendimiento real de cada sucursal.